python/GafferSceneUI/AttributeQueryUI.py

```diff
@@ -36,6 +36,75 @@
 
 import Gaffer
 import GafferScene
+import GafferUI
+import imath
+import IECore
+import functools
+import uuid
+
+from GafferSceneUI._GafferSceneUI import __setupMenuTitle as setupMenuTitle
+from GafferSceneUI._GafferSceneUI import __setupMenuNames as setupMenuNames
+from GafferSceneUI._GafferSceneUI import __setupFromMenuName as setupFromMenuName
+
+class _SetupButton( GafferUI.Widget ) :
+
+	def __init__( self, node ) :
+
+		self.__node = node
+		self.__row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal )
+
+		GafferUI.Widget.__init__( self, self.__row )
+
+		with self.__row :
+
+				GafferUI.Spacer( imath.V2i( GafferUI.PlugWidget.labelWidth(), 1 ) )
+
+				menuButton = GafferUI.MenuButton(
+					image = "plus.png",
+					hasFrame = False,
+					menu = GafferUI.Menu(
+						Gaffer.WeakMethod( self.__menuDefinition ),
+						title = setupMenuTitle()
+					)
+				)
+
+				GafferUI.Spacer( imath.V2i( 1 ), imath.V2i( 999999, 1 ), parenting = { "expand" : True } )
+
+		node.childAddedSignal().connect( Gaffer.WeakMethod( self.__updateVisibility ), scoped = False )
+		node.childRemovedSignal().connect( Gaffer.WeakMethod( self.__updateVisibility ), scoped = False )
+
+		self.__updateVisibility()
+
+	def __menuDefinition( self ) :
```

Comment:
we would need to pass the attributequery node, which is not a problem. However the c++ plugadder class has a menusignal() function that returns a boost python signal that is connected to on the python side to actually create an IECore.MenuDefinition from a list of strings. there is no such signal with a ```Button``` in python.

Reply:
I was thinking something like this :

```
def __init__( self ) :

    ...
    # Instead of MenuButton
    button = GafferUI.Button()
    button.clickedSignal().connect( Gaffer.WeakMethod( self.__clicked ) )
    ...

def __clicked( self, button ) :

    # Call the C++ function which uses `PlugAdder.menuSignal()` to do its thing.
    # (this would replace the three functions we currently have and the duplicate menu code).
    GafferSceneUI.AttributeQueryUI.showSetupMenu( self.__node )
```

I think this would work without needing to change PlugAdder?

Reply:
Just noticed that PlugAdder::menuSignal() is static so we don't need a PlugAdder instance, so yes this is possible